osf/models/user.py

```diff
@@ -750,10 +750,10 @@ def merge_user(self, user):
             for key, value in user.mailchimp_mailing_lists.items():
                 # subscribe to each list if either user was subscribed
                 subscription = value or self.mailchimp_mailing_lists.get(key)
-                signals.user_merged.send(self, list_name=key, subscription=subscription)
+                signals.user_update_mailchimp_subscription.send(self, list_name=key, subscription=subscription)
```

Comment:
refactored `user_merged` to `user_account_merged` for better naming uniformity `user_account_re/deactivated`  with and to split up this into a more sensible signal name.

Reply:
instead of adding a mailchimp-specific signal, might be an improvement to move this whole mailchimp-specific loop into something that listens for `user_account_merged`